### PR TITLE
fix: GC `parquet_file` table in batches

### PR DESCRIPTION
**Pull request by crepererum**
Otherwise this transaction will never finish in prod.

### CODE REVIEW

File: garbage_collector/src/parquetfile/deleter.rs [hunk 1]
```diff
@@ -23,11 +23,13 @@ pub(crate) async fn perform(
             .context(DeletingSnafu)?;
         info!(delete_count = %deleted.len(), "iox_catalog::delete_old()");
 
-        select! {
-            _ = shutdown.recv() => {
-                break
-            },
-            _ = sleep(Duration::from_secs(60 * sleep_interval_minutes)) => (),
+        if deleted.is_empty() {
```
Comment (alamb) [inline garbage_collector/src/parquetfile/deleter.rs:26]: Does this change mean the garbage collector won't respond to shutdowns until it has no files to delete?

File: iox_catalog/src/postgres.rs [hunk 2]
```diff
@@ -1719,10 +1719,17 @@ RETURNING *;
     }
 
     async fn delete_old_ids_only(&mut self, older_than: Timestamp) -> Result<Vec<ParquetFileId>> {
+        // see https://www.crunchydata.com/blog/simulating-update-or-delete-with-limit-in-postgres-ctes-to-the-rescue
         let deleted = sqlx::query(
             r#"
+WITH parquet_file_ids as (
```
Comment (alamb) [inline iox_catalog/src/postgres.rs:1725]: 👍

File: iox_catalog/src/postgres.rs [hunk 3]
```diff
@@ -1719,10 +1719,17 @@ RETURNING *;
...
+    SELECT id
+    FROM parquet_file
+    WHERE to_delete < $1
+    LIMIT 1000
```
Comment (alamb) [inline iox_catalog/src/postgres.rs:1729]: Is it worth making this 1000 a named constant somewhere?

**Review by alamb — APPROVED** [review]
makes sense to me. I don't really have any good ideas for testing sadly

File: garbage_collector/src/parquetfile/deleter.rs (hunk 1 above)
Comment (crepererum) [reply garbage_collector/src/parquetfile/deleter.rs:27]: good catch, fixed

File: iox_catalog/src/postgres.rs (hunk 3 above)
Comment (crepererum) [reply iox_catalog/src/postgres.rs]: done

**Comment by lukebond** [comment]
:+1: thank you @crepererum

**Pull request merged**